packages/datadog-shimmer/src/shimmer.js

```diff
@@ -73,25 +89,67 @@ function wrap (target, name, wrapper) {
     }
     descriptor.value = wrapped
   } else {
-    if (descriptor.get || descriptor.set) {
-      // TODO(BridgeAR): What happens in case there is a setter? This seems wrong?
-      // What happens in case the user does indeed set this to a different value?
-      // In that case the getter would potentially return the wrong value?
-      descriptor.get = () => wrapped
+    if (descriptor.get) {
+      // replaceGetter may only be used when the getter has no side effect.
+      if (options?.replaceGetter) {
+        if (descriptor.set) {
+          // This case is possible by replacing the setter with a wrapper that
+          // either undos the original get replacement (that should be safe to
+          // do, we just loose the instrumentation afterwards) or that
+          // reinstruments the passed through value after calling get. The
+          // latter might however have side effects due to maybe not replacing
+          // the value that the getter instrumented.
+          throw new Error('Replacing getter with sets is not supported. Implement if required.')
+        }
+        descriptor.get = () => wrapped
+      } else {
+        descriptor.get = wrapped
```

Comment:
Just a mental note to help me understand the flow (maybe it warrants a code comment): We can only get here if `replaceGetter: false` and the property has a getter. There might also be a setter. In this case `wrapped` is a product of `wrapper(descriptor.get)`.

If you want to add a comment, maybe this one:

```suggestion
        // State: Property has a getter and `replaceGetter: false`.
        // The `wrapped` property is a product of `wrapper(descriptor.get)`
        descriptor.get = wrapped
```

Reply:
I restructured the code a bit and I believe it is now easier to follow in general.